app/src/main/java/com/gcect/gcectapp/ui/fragments/HomeFragment.kt

```diff
@@ -228,6 +240,55 @@ class HomeFragment : Fragment(), OnHomeNoticeItemClickListener, OnBottomSheetIte
                     pd.cancel()
                 }
             })
+
+    }
+
+    fun createNotificationChannel() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val channelId = "all_notifications" // You should create a String resource for this instead of storing in a variable
+            val mChannel = NotificationChannel(
+                channelId,
+                "General Notifications",
+                NotificationManager.IMPORTANCE_DEFAULT
+            )
+            mChannel.description = "This is default channel used for all other notifications"
+
+            val notificationManager =
+                requireContext().getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+            notificationManager.createNotificationChannel(mChannel)
+        }
+    }
+
+    private fun showNotification(progress: Long){
```

Comment:
Creating an extension function will be better choice , So we can reuse this in other fragments also

Reply:
How to make that one I'm new to Kotlin that's why.